src/hangar/cli/cli.py

```diff
@@ -535,136 +536,172 @@ def server(overwrite, ip, port, timeout):
+        if not os.path.isdir(outdir):
+            click.echo(f"Directory {outdir} does not exist")
+            return
 
-        with arrayset as aset, click.progressbar(sampleNames) as sNamesBar:
+        with aset, click.progressbar(sampleNames) as sNamesBar:
             for sampleN in sNamesBar:
-                if format_:
-                    if sampleN.endswith(format_):
-                        outFP = os.path.join(outP, f'{sampleN}')
-                    else:
-                        outFP = os.path.join(outP, f'{sampleN}.{format_}')
-                else:
-                    outFP = os.path.join(outP, f'{sampleN}')
                 try:
                     data = aset[sampleN]
-                    imsave(outFP, data)
+                    external.save(data, outdir, sampleN, extension, plugin, **kwargs)
                 except KeyError as e:
                     click.echo(e)
     finally:
         co.close()
 
 
-@main.command(name='view')
-@click.argument('startpoint', nargs=1, type=str, required=True)
+@main.command(name='view', context_settings=dict(allow_extra_args=True, ignore_unknown_options=True,))
 @click.argument('arrayset', type=str, required=True)
 @click.argument('sample', type=str, required=True)
-@click.option('--plugin', default=None,
-              help='Plugin name to use instead of auto-inferred plugin')
+@click.argument('startpoint', nargs=1, default=None, required=False)
+@click.option('-f', '--format', 'format_', required=False, help='File format of output file')
+@click.option('--plugin', default=None, help='Plugin name to use instead of auto-inferred plugin')
 @pass_repo
-def view_data(repo: Repository, startpoint, arrayset, sample, plugin):
+@click.pass_context
+def view_data(ctx, repo: Repository, arrayset, sample, startpoint, format_, plugin):
     """Use a plugin to view the data of some SAMPLE in ARRAYSET at STARTPOINT.
     """
+    # TODO: ignore warning through env variable
     from hangar.records.commiting import expand_short_commit_digest
-    from hangar.records.heads import get_branch_head_commit
-    from hangar.cli.io import imshow, show
+    from hangar.records.heads import get_branch_head_commit, get_staging_branch_head
+    from hangar import external
 
+    kwargs = parse_custom_arguments(ctx.args)
     if startpoint in repo.list_branches():
         base_commit = get_branch_head_commit(repo._env.branchenv, startpoint)
-    else:
+    elif startpoint:
         base_commit = expand_short_commit_digest(repo._env.refenv, startpoint)
+    else:
+        branch_name = get_staging_branch_head(repo._env.branchenv)
+        base_commit = get_branch_head_commit(repo._env.branchenv, branch_name)
+    co = repo.checkout(commit=base_commit)
+
+    sample, stype = sample.split(':') if ':' in sample else (sample, '')
+    sample = int(sample) if stype == 'int' else str(sample)
+
+    extension = format_.lstrip('.') if format_ else None
 
     try:
```

Comment:
The repo was checked out way to early here, if the code between it and the `try` block every raises an error, the program would never get to the point where it would close.